packages/cli-kit/src/private/node/ui/components/SelectInput.tsx

```diff
@@ -161,7 +163,14 @@ function SelectInputInner<T>(
   }
 
   const hasAnyGroup = rawItems.some((item) => typeof item.group !== 'undefined')
-  const items = sortBy(rawItems, 'group')
+  const items = sortBy(rawItems, [
+    (item) => {
+      if (!groupOrder || !item.group) return Number.MAX_SAFE_INTEGER
+      const index = groupOrder.indexOf(item.group)
+      return index === -1 ? Number.MAX_SAFE_INTEGER : index
+    },
+    'group',
+  ])
   const itemsHaveKeys = items.some((item) => typeof item.key !== 'undefined' && item.key.length > 0)
 
```

Comment:
Maybe we can also add a test in `SelectInput.test.tsx` to validate that the group order is working correctly?